File: app/src/main/java/com/kode/utair/data/repository/WeatherSettingsRepository.java
package com.kode.utair.data.repository;

import com.kode.utair.data.repository.datasource.WeatherPreferences;
import com.kode.utair.domain.models.WeatherSettings;
import com.kode.utair.domain.repository.IWeatherSettingsRepository;

import javax.inject.Inject;

import io.reactivex.Completable;
import io.reactivex.Single;

public class WeatherSettingsRepository implements IWeatherSettingsRepository {

    private WeatherPreferences weatherPreferences;

    @Inject
    public WeatherSettingsRepository(WeatherPreferences weatherPreferences) {

        this.weatherPreferences = weatherPreferences;
    }

    @Override
    public Single<WeatherSettings> getWeatherSettings() {
        return Single.create(e -> {
            e.onSuccess(weatherPreferences.getWeatherSettings());
        });
    }

    @Override
    public Completable saveWeatherSettings(WeatherSettings weatherSettings) {
        return Completable.create(e -> {
            weatherPreferences.saveWeatherSettings(weatherSettings);
            e.onComplete();
        });
    }

}

File: app/src/main/java/com/kode/utair/data/repository/WeatherSettingsRepository.kt
package com.kode.utair.data.repository

import com.kode.utair.data.repository.datasource.WeatherPreferences
import com.kode.utair.domain.models.WeatherSettings
import com.kode.utair.domain.repository.IWeatherSettingsRepository
import io.reactivex.Completable
import io.reactivex.CompletableEmitter
import io.reactivex.Single
import io.reactivex.SingleEmitter
import javax.inject.Inject

class WeatherSettingsRepository @Inject constructor(
        private val weatherPreferences: WeatherPreferences
) : IWeatherSettingsRepository {

    override fun getWeatherSettings(): Single<WeatherSettings> {
        return Single.create { e: SingleEmitter<WeatherSettings> ->
            e.onSuccess(weatherPreferences.getWeatherSettings())
        }
    }

    override fun saveWeatherSettings(weatherSettings: WeatherSettings): Completable {
        return Completable.create { e: CompletableEmitter ->
            weatherPreferences.saveWeatherSettings(weatherSettings)
            e.onComplete()
        }
    }

}